src/table/src/Shared/Table.lua

```diff
@@ -200,16 +197,7 @@ end
 	@return The index of the value, if found
 	@return nil -- if not found
 ]=]
-function Table.getIndex(haystack, needle)
-	assert(needle ~= nil, "Needle cannot be nil")
-
-	for index, item in pairs(haystack) do
-		if needle == item then
-			return index
-		end
-	end
-	return nil
-end
+Table.getIndex = table.find
```

Comment:
This may actually be a breaking change, does table.find respect metatableS? probably still fine. may need to increment version of table library though.

Reply:
I suspect we should major-version this package.

Reply:
Good catch. Looks like it does, here's a snippet.
```lua
local function getIndex(haystack, needle)
	for i, v in pairs(haystack) do
		if needle == v then
			return i
		end
	end
end

local mt = {}
mt.__eq = function(a, b)
	return true
end

local a = setmetatable({}, mt)
local b = setmetatable({}, mt)

local testArr = {a, b}

-- 1, 1.
print(getIndex(testArr, b), table.find(testArr, b))
-- 2, 2.
mt.__eq = nil
print(getIndex(testArr, b), table.find(testArr, b))
```